migrations/2018_06_27_101500_change_flags_b8_columns.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        // Make sure the entities exist so that we will be able to create
+        // foreign keys without any issues.
+        $connection = $schema->getConnection();
+        $prefix = $connection->getTablePrefix();
+        $connection->statement("delete from {$prefix}flags where not exists (select 1 from {$prefix}posts where id = post_id)");
```

Comment:
We should really stick to Eloquent as much as possible. These raw queries will bite us in the long run with cross database driver compatibility.

Do you want me to refactor this?

Reply:
Think we already spoke about this on discord, but for the record: Maybe we should refactor them but it's not a priority as we can do so at any time in the future. (Also they are pretty standard SQL so unlikely to cause too many issues)

Priority would be to make sure the statements are split up into separate migrations appropriately so that if a statement fails it doesn't break

Reply:
I looked into this and these flag statements can be rewritten. Those from tags cannot: https://github.com/flarum/flarum-ext-tags/pull/53/files

as mentioned on discord.

I'm starting to feel stronger to keeping them as is..